pkg/api/v1/helpers.go

```diff
@@ -480,3 +486,89 @@ type NodeResources struct {
 	// Capacity represents the available resources of a node
 	Capacity ResourceList `protobuf:"bytes,1,rep,name=capacity,casttype=ResourceList,castkey=ResourceName"`
 }
+
+// Tries to add a taint to annotations list. Returns a new copy of updated Node and true if something was updated
+// false otherwise.
+func AddOrUpdateTaint(node *Node, taint *Taint) (*Node, bool, error) {
```

Comment:
@timothysc If you mean modifying annotations stuff in this function to api fields, then ofcourse, its already done in my PR. Or are you asking something else?